### PR TITLE
docs: correct errors in timeout policy section

### LINKED ISSUE #2544: HTTPProxy timeoutPolicy.response of "0s" (as mentioned in docs) is NOT infinite
With contour 1.4.0, I've just attempted to set a timeoutPolicy.response, and "0s" doesn't seem to work.

I expected the value "0s" to mean an infinite timeout, as the docs describe:
> `timeoutPolicy.response` This field can be any positive time period or “infinity”. The time period of 0s will also be treated as infinity. This timeout covers the time from the end of the client request to the end of the upstream response. By default, Envoy has a 15 second value for this timeout. More information can be found in Envoy’s documentation.
https://projectcontour.io/docs/v1.4.0/httpproxy/

First with `"infinity"` (as worked in the old Ingress [annotation](https://projectcontour.io/docs/v1.4.0/annotations/)):
```
      timeoutPolicy:
        response: "infinity"
        idle: "355s"
```

That gives a timeout of zero in Envoy's `/config_dump`:
```
          "route": {
           "cluster": "default/myapp-service/5000/da33a3ff5e",
           "timeout": "0s",
           "idle_timeout": "355s"
          }
```

But then with `"0s"` (as described in the `timeoutPolicy` documentation):
```
      timeoutPolicy:
        response: "0s"
        idle: "355s"
```

That gives a default setting (no timeout field) in Envoy's `/config_dump`:
```
          "route": {
           "cluster": "default/myapp-service/5000/da33a3ff5e",
           "idle_timeout": "355s"
          }
```

So I have a workaround (setting timeoutPolicy.response to "infinity"), but either there's a bug or the docs are out of date.  😺

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <krisss@vmware.com>

closes #2544

### CODE REVIEW

File: site/docs/master/httpproxy.md [hunk 1]
```diff
@@ -673,14 +673,15 @@ In this example, requests to `timeout.bar.com/` will have a response timeout pol
 This refers to the time that spans between the point at which complete client request has been processed by the proxy, and when the response from the server has been completely processed.
 
 - `timeoutPolicy.response` This field can be any positive time period or "infinity".
-The time period of **0s** will also be treated as infinity.
 This timeout covers the time from the *end of the client request* to the *end of the upstream response*.
 By default, Envoy has a 15 second value for this timeout.
 More information can be found in [Envoy's documentation][4].
+Note that a value of **0s** will be treated as if the field were not set, i.e. by using Envoy's default behavior.
```
Comment (jpeach) [inline site/docs/master/httpproxy.md:679]: Can you please also update the API documentation in `apis/projectcontour/v1/httpproxy.go`, which is the authoritative reference for the HTTPProxy spec?

File: site/docs/master/httpproxy.md [hunk 2]
```diff
@@ -1416,3 +1417,4 @@ Some examples of invalid configurations that Contour provides statuses for:
  [9]: {% link docs/master/annotations.md %}
  [10]: /docs/{{site.latest}}/api/#projectcontour.io/v1.Service
  [11]: configuration.md#fallback-certificate
+ [12]: https://www.envoyproxy.io/docs/envoy/v1.11.2/api-v2/api/v2/route/route.proto.html#envoy-api-field-route-routeaction-idle-timeout
```
Comment (jpeach) [inline site/docs/master/httpproxy.md:1420]: It probably makes sense to use a versioned link here, but we should link to the latest version `1.14.2`.

File: site/docs/master/httpproxy.md (hunk 1 above)
Comment (skriss) [reply site/docs/master/httpproxy.md:679]: added the same note to the API docs

File: site/docs/master/httpproxy.md (hunk 2 above)
Comment (skriss) [reply site/docs/master/httpproxy.md]: updated all the links on the page; I also realized there was already a link for the idle timeout (`[6]`) so de-duped.

File: apis/projectcontour/v1/httpproxy.go [hunk 3]
```diff
@@ -297,6 +297,7 @@ type TCPHealthCheckPolicy struct {
 // TimeoutPolicy durations are expressed in the Go [Duration format](https://godoc.org/time#ParseDuration).
 // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
 // The string "infinity" is also a valid input and specifies no timeout.
+// A value of "0s" will be treated as if the field were not set, i.e. by using Envoy's default behavior.
```
Comment (jpeach) [inline apis/projectcontour/v1/httpproxy.go:300]: There's 2 comments on the struct fields about the timeout ... are they still accurate?

Comment (skriss) [reply apis/projectcontour/v1/httpproxy.go:300]: I added some more clarity there around default behavior.

**Pull request merged**